scripts/parse_tools/parse_checkers.py

```diff
@@ -936,6 +936,55 @@ def check_diagnostic_id(test_val, prop_dict, error):
 
 ########################################################################
 
+def check_molar_mass(test_val, prop_dict, error):
+    """Return <test_val> if valid molar mass, otherwise, None
+    if <error> is True, raise an Exception if <test_val> is not valid.
+    >>> check_molar_mass('1', None, True)
+    1.0
+    >>> check_molar_mass('1.0', None, True)
+    1.0
+    >>> check_molar_mass('1.0', None, False)
+    1.0
+    >>> check_molar_mass('-1', None, False)
+
+    >>> check_molar_mass('-1.0', None, False)
+
+    >>> check_molar_mass('string', None, False)
+
+    >>> check_molar_mass('-1', None, True) #doctest: +IGNORE_EXCEPTION_DETAIL
+    Traceback (most recent call last):
+    CCPPError: '-1' is not a valid molar mass
+    >>> check_molar_mass('-1.0', None, True) #doctest: +IGNORE_EXCEPTION_DETAIL
+    Traceback (most recent call last):
+    CCPPError: '-1.0' is not a valid molar mass
+    >>> check_molar_mass('string', None, True) #doctest: +IGNORE_EXCEPTION_DETAIL
+    Traceback (most recent call last):
+    CCPPError: '-1.0' is not a valid molar mass
+    """
+    # Check if input value is an int or float
+    try:
+        valid = isinstance(float(test_val), float)
```

Comment:
Why call `float` twice? Also, if `float` returns it *must* be an instance of `float`.
```suggestion
        test_val = float(test_val)
        if (test_val > 0.0) and (test_val < float('inf')):
            return None
        # end if
        if error:
           raise CCPPError(f"{test_val} is not a valid molar mass")
        # end if
```
In writing this, I was wondering if we should have a maximum allowed value. I put in `'inf'` which is an obvious max (only caught if someone tries `molar_mass = "inf"`). If you decide to try this, I suggest putting a module constant at the top of the file (e.g., `_MAX_MOLAR_MASS = float('inf')` or `_MAX_MOLAR_MASS = 10000.0`) and using that for the test.

Reply:
added upper limit and tests.